### ISSUE TITLE
make the `connect()` method more like `net.Socket.connect()`

### ISSUE THREAD

**Issue by KevinRamharak** [dex4er/js-promise-socket#3]
Hi,

I was looking for promise based socket class and it seems i found it. I was just wondering why the `connect()` method does not look like the original signature.

the current signature is
```ts
connect(port: number): Promise<void>;
connect(path: string): Promise<void>;
connect(options: object): Promise<void>;
```
I would propose something like this for `PromiseSocket`
```ts
connect(port: number, host?: string): Promise<this>;
connect(path: string): Promise<this>;
connect(options: net.SocketConnectOpts): Promise<this>; // or just 'object'
```

Let me know what you think.

**Comment by dex4er** [comment]
I never seen `Promise<this>`. Probably it doesn't make sense. But of course `connect` might have additional arguments, like `host` and `options` might be the same as in `@types/node` for `net` module.

Thanks for your report, I'll change this method.

**Comment by dex4er** [comment]
Fixed in v3.1.0

**Comment by KevinRamharak** [comment]
Thanks,
Not sure if `Promise<this>` does really add anything. I just taught it would mimic the real API, but then again that was to help the user to chain methods calls on each other. With a more synchronous API it is not really usefull

I noticed you changed the `.d.ts` file, but the implementation will ignore the `host` part with this signature:
```ts
connect(port: number, host?: string): Promise<this>;
```

This is because the implementation states:
```js
connect(options) {
  // ...
  socket.connect(options, handler)
}
```
I would propose something like:
```js
connect(...options) {
  // ...
  socket.connect(...options, handler)
}
```

**Comment by dex4er** [comment]
OMG, I didn't commit all my changes: only those for `d.ts` file :(

**Comment by dex4er** [comment]
Fixed in v3.1.1

**Comment by KevinRamharak** [comment]
super! thank you
